File: src/main/java/com/elchologamer/userlogin/listener/restriction/ReceiveDamageRestriction.java
package com.elchologamer.userlogin.listener.restriction;

import org.bukkit.event.EventHandler;
import org.bukkit.event.entity.EntityDamageEvent;

public class ReceiveDamageRestriction extends BaseRestriction<EntityDamageEvent> {

    public ReceiveDamageRestriction() {
        super("damage.receive");
    }

    @EventHandler
    public void handle(EntityDamageEvent e) {
        if (shouldRestrict(e)) e.setCancelled(true);
    }
}

File: src/main/kotlin/com/elchologamer/userlogin/listener/restriction/ReceiveDamageRestriction.kt
package com.elchologamer.userlogin.listener.restriction

import org.bukkit.event.EventHandler
import org.bukkit.event.entity.EntityDamageEvent

object ReceiveDamageRestriction : BaseRestriction<EntityDamageEvent>("damage.receive") {
    @EventHandler
    fun handle(e: EntityDamageEvent) {
        if (shouldRestrict(e)) e.isCancelled = true
    }
}